packages/medusa/src/api/routes/admin/reservations/list-reservations.ts

```diff
@@ -110,12 +116,60 @@ import {
 export default async (req: Request, res: Response) => {
   const inventoryService: IInventoryService =
     req.scope.resolve("inventoryService")
+  const manager: EntityManager = req.scope.resolve("manager")
+
+  const { filterableFields, listConfig } = req
+
+  const relationsPredicates: ((string) => boolean)[] = []
+
+  // join item
+  const includeItems = !!listConfig.relations?.includes("line_item")
+
+  if (includeItems) {
+    relationsPredicates.push((r) => r !== "line_item")
+  }
+
+  const includeInventoryItems =
+    !!listConfig.relations?.includes("inventory_item")
+
+  if (includeInventoryItems) {
+    relationsPredicates.push((r) => r !== "inventory_item")
+  }
+
+  if (relationsPredicates.length) {
+    listConfig.relations = listConfig.relations?.filter((r) =>
+      relationsPredicates.every((p) => p(r))
+    )
+  }
```

Comment:
see https://github.com/medusajs/medusa/pull/4115#discussion_r1200138785 but there is an overlap between the pr's so I just post the link and will try to not re comment what I said on the other pr, let me know wdyt

Reply:
should be fixed here, something with updating branches probably messed up the diff as it should only be the 9 files it is now 😄